src/he/wolf.c

```diff
@@ -38,9 +38,9 @@ int he_wolf_dtls_read(WOLFSSL *ssl, char *buf, int sz, void *ctx) {
   // Get DTLS context
   he_conn_t *conn = (he_conn_t *)ctx;
 
-  // Abort if incoming_data is null - this is set internally in libhelium
+  // This can be null if no data has been received yet, tell wolfSSL to stop asking
   if(!conn->incoming_data) {
-    return WOLFSSL_CBIO_ERR_GENERAL;
+    return WOLFSSL_CBIO_ERR_WANT_READ;
```

Comment:
Good catch, @expressvpn-raihaan-m. I think this is a regression caused by #26 (https://github.com/expressvpn/lightway-core/pull/26/files#diff-6668aae42fdd94c96080359c852e7f6c578583b8508292c998a6d077345f713eR41-R45).